cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -629,6 +629,38 @@ TEST_F(GcsIntegrationTest, GetFileInfoBucket) {
   ASSERT_RAISES(Invalid, fs->GetFileInfo("gs://" + PreexistingBucketName()));
 }
 
+// We intentionally do not test invalid permissions with storage-testbench,
+// because the testbench does not enforce any permission checks (ACL/IAM).
+// See:
+// https://github.com/googleapis/storage-testbench?tab=readme-ov-file#what-is-this-testbench
+//
+// In real GCS environments, trying to access a bucket without permission
+// results in:
+//   - Error code: 5
+//   - Status    : NOT_FOUND
+//
+// The following test depends on real GCS access and is not suitable for CI/CD
+// environments. To run this test manually, set the environment variable:
+//     ARROW_RUN_REAL_GCS_TESTS=1
+//
+// Example:
+//     ARROW_RUN_REAL_GCS_TESTS=1 ./debug/arrow-gcsfs-test
+TEST_F(GcsIntegrationTest, GetFileInfoWithoutPermission) {
+  if (!std::getenv("ARROW_RUN_REAL_GCS_TESTS")) {
```

Comment:
Could you use `ARROW_TEST_` like https://github.com/apache/arrow/blob/596f156f0c4df8087a1ecc1d9388fc2955461719/cpp/src/arrow/filesystem/s3fs_benchmark.cc#L54-L57 something like?

```suggestion
  if (!std::getenv("ARROW_TEST_GCS_USE_REAL_SERVICE")) {
```

Reply:
I wasn't aware of that convention. I appreciate you fixing it!